### PR TITLE
Payments/merge charge with capture

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because...

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

File: docs/guides/payments.rst [hunk 1]
```diff
@@ -39,13 +39,13 @@ Example
 
 .. note::
 
-    All the below methods receive ``payment_information`` as a dataclass: ``PaymentData``. 
-    Methods should return a response as a dataclass: ``GatewayResponse``. 
+    All the below functions receive ``payment_information`` as a dataclass: ``PaymentData`` and ``config`` as a dataclass: ``ConfigData``. 
+    Functions should return a response as a dataclass: ``GatewayResponse``.
```
Comment (NyanKiyoshi) [inline docs/guides/payments.rst]:
```suggestion
    Those functions should return a response as a dataclass: ``GatewayResponse``. 
```

File: docs/guides/payments.rst [hunk 2]
```diff
@@ -419,6 +443,9 @@ Please take a moment to consider the example settings above.
 - ``connection_params``
     List of parameters used for connecting to the payment's gateway.
 
+- ``auto_capture``
+    Define if the gateway should also capture funds from the card. When auto_capture is set to False, funds will be blocked but manual capture will be required.
```
Comment (NyanKiyoshi) [inline docs/guides/payments.rst]:
```suggestion
    Define if the gateway should also capture funds from the card. When ``auto_capture`` is set to ``False``, funds will be blocked by the customer's bank for a 7-days period, a manual capture will be required.
```

File: saleor/payment/models.py [hunk 3]
```diff
@@ -154,11 +154,14 @@ def can_authorize(self):
         return self.is_active and self.not_charged
 
     def can_capture(self):
-        return self.is_active and self.not_charged and self.is_authorized
+        if not (self.is_active and self.not_charged):
+            return False
+
+        gateway_config = settings.PAYMENT_GATEWAYS[self.gateway]["config"]
```
Comment (maarcingebala) [inline saleor/payment/models.py]: Shouldn't we use `get_payment_gateway` here?

File: docs/guides/payments.rst [hunk 4]
```diff
@@ -402,15 +416,19 @@ Adding new payment gateway to the settings
 .. code-block:: python
 
     PAYMENT_GATEWAYS = {
-        "braintree": {
+        'braintree': {
```
[the remark this answers is not in the document]
Comment (korycins) [reply docs/guides/payments.rst]: My bad. Thanks

File: docs/guides/payments.rst [hunk 5]
```diff
@@ -232,7 +226,7 @@ Parameters
 +-------------------------+-----------------+-----------------------------------------------------------------------------+
 | ``payment_information`` | ``PaymentData`` | Payment information, containing the token, amount, currency and billing.    |
 +-------------------------+-----------------+-----------------------------------------------------------------------------+
-| ``connection_params``   | ``dict``        | List of parameters used for connecting to the payment's gateway.            |
+| ``config``              | ``ConfigData``  | Configuration of the payment gateway.                                               |
```
Comment (akjanik) [inline docs/guides/payments.rst]: nit: isn't last pipe (`|`) too far?

File: docs/guides/payments.rst [hunk 6]
```diff
@@ -286,6 +280,19 @@ AddressData
...
 
+ConfigData
+""""""""""
+
++---------------------+-----------+---------------------------------------------------------------------------------------------------------+
+| name                | type      | description                                                                                             |
++---------------------+-----------+---------------------------------------------------------------------------------------------------------+
+| auto_capture        | ``bool``  | Define if gateway should also capture funds from the card. If false, payment should be only authorized  |
++---------------------+-----------+---------------------------------------------------------------------------------------------------------+
+| template_path       | ``str``   | Should specify a path to a template that will be rendered for the checkout.                             |
```
Comment (akjanik) [inline docs/guides/payments.rst]:
```suggestion
| template_path       | ``str``   | Path to a template that will be rendered for the checkout.                             |
```

File: docs/guides/payments.rst (hunk 5 above)
Comment (NyanKiyoshi) [reply docs/guides/payments.rst]: Looks like it

Screenshot 2019-05-14 at 12 21 02

**Pull request merged**